addons/mail/static/src/core/message_model.js

```diff
@@ -62,6 +62,11 @@ export class Message {
     type;
     /** @type {string} */
     temporary_id = null;
+    /** @type {string|undefined} */
+    notificationType;
```

Comment:
`notification` name already has a lot of different meaning in this context, I would avoid using it here.

Considering the code in `notificationIcon`, isn't it useless, just "is pinned" (pinned_at)?

Reply:
The notification itself is not pinned, so that would not work.